### PR TITLE
[WIP][DCOS-50943] Added additional settings in Elastic config

### PR DESCRIPTION
**Pull request by rishabh96b**
1. xpack.monitoring.enabled
2. xpack.ml.enabled
3. xpack.watcher.enabled
4. xpack.monitoring.enabled

### CODE REVIEW

File: frameworks/elastic/src/main/dist/elasticsearch.yml [hunk 1]
```diff
@@ -53,6 +53,10 @@ xpack.security.transport.ssl.enabled: true
 # Client encryption
 xpack.security.http.ssl.enabled: true
 xpack.security.http.ssl.client_authentication: none
+xpack.monitoring.enabled: {{XPACK_MONITORING_ENABLED}}
+xpack.ml.enabled: {{XPACK_ML_ENABLED}}
+xpack.watcher.enabled: {{XPACK_WATCHER_ENABLED}}
+xpack.reporting.enabled: {{XPACK_REPORTING_ENABLED}}
```
Comment (mpereira) [inline frameworks/elastic/src/main/dist/elasticsearch.yml]: Why are these settings in the `SECURITY_TRANSPORT_ENCRYPTION_ENABLED=true` block?

File: frameworks/elastic/universe/config.json [hunk 2]
```diff
@@ -1060,6 +1060,26 @@
...
+        },
+        "xpack_watcher_enabled": {
+          "description": "X-Pack ml is enabled by default on Elastic 6.3+. Changing this setting has no effect.",
+          "type": "boolean",
+          "default": true
+        },
+        "xpack_reporting_enabled": {
+          "description": "X-Pack Monitoring us enabled by default on Elastic 6.3+. Changing this setting has no effect.",
+          "type": "boolean",
+          "default": true
```
Comment (mpereira) [inline frameworks/elastic/universe/config.json]: The descriptions can be improved. Also, make sure that the default values for our settings mirror those of Elasticsearch. For example, it looks like X-Pack monitoring is disabled by default on Elasticsearch.

File: frameworks/elastic/src/main/dist/elasticsearch.yml (hunk 1 above)
Comment (rishabh96b) [reply frameworks/elastic/src/main/dist/elasticsearch.yml]: All the xpack related settings are defined here so I thought of including these too. Should I create separate block for these additional settings?

File: frameworks/elastic/universe/config.json (hunk 2 above)
Comment (rishabh96b) [reply frameworks/elastic/universe/config.json]: Yes, AFAIK the settings that I have included are enabled by default.
Please have a look here: https://www.elastic.co/guide/en/x-pack/current/xpack-settings.html

Comment (mpereira) [reply frameworks/elastic/universe/config.json]: It looks like the default values for all but `xpack.reporting.enabled` is `true`:
```
$ dcos task exec elastic__master-0-node__e2d370c5-d916-48c1-8630-7e4f8fbfe3f5 /opt/mesosphere/bin/curl -ki -H 'Content-type: application/json' 'http://coordinator.elastic.l4lb.thisdcos.directory:9200/_cluster/settings?include_defaults=true&format=yaml&flat_settings=true' | grep 'xpack.monitoring.enabled\|xpack.ml.enabled\|xpack.watcher.enabled\|xpack.monitoring.enabled'
  xpack.ml.enabled: "true"
  xpack.monitoring.enabled: "true"
  xpack.watcher.enabled: "true"
```

`xpack.reporting.enabled` seems to be a Kibana-only setting, so we can remove it from here.

Comment (rishabh96b) [reply frameworks/elastic/universe/config.json]: Okay, let's remove it. Thanks :)

Comment (rishabh96b) [reply frameworks/elastic/universe/config.json]: Hey @mpereira !Can you have a look at it?

File: frameworks/elastic/universe/marathon.json.mustache [hunk 3]
```diff
@@ -106,6 +106,7 @@
     "ELASTICSEARCH_HEALTH_USER": "{{elasticsearch.health_user}}",
     "ELASTICSEARCH_HEALTH_USER_PASSWORD": "{{elasticsearch.health_user_password}}",
     "TASKCFG_ALL_XPACK_SECURITY_ENABLED": "{{elasticsearch.xpack_security_enabled}}",
+    "TASK_CFG_ALL_XPACK_SECURITY_AUDIT_ENABLED": "{{elasticsearch.xpack_security_audit_enabled}}",
```
Comment (mpereira) [inline frameworks/elastic/universe/marathon.json.mustache]: The prefix here should be `TASKCFG` instead of `TASK_CFG`.

**Pull request closed without merging**